File: Java/remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int position = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[position] = nums[i];
                position++;
            } 
        }
        return position;
    }
}

File: Kotlin/remove-element.kt
class Solution {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        var position = 0
        for (index in nums.indices) {
            if (nums[index] != `val`) {
                nums[position] = nums[index]
                position++
            }
        }
        return position
    }
}
